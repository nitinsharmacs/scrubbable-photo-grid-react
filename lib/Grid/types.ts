import type { GridConfigType, SectionType } from 'lib/types';

export interface GridProps {
  gridData: SectionType[];
  config: GridConfigType;
}
