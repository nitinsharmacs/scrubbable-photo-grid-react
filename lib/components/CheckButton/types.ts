import type { CSSProperties } from 'react';

export interface CheckButtonProps {
  active?: boolean;
  label: string;
  id: string;
  styles?: CSSProperties;
  onClick?: (e: React.SyntheticEvent) => void;
}
