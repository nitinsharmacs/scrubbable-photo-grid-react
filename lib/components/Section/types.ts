import type { SegmentCtx } from 'lib/components/Segment/types';
import type { GridConfigType, SectionConfigType, SectionType } from 'lib/types';

export type SectionSelectionHandler = (
  sectionId: string,
  ctx: SectionCtx
) => void;

export interface SectionProps {
  section: SectionType;
  config: GridConfigType;
  map: SectionConfigType;
  onSelect?: SectionSelectionHandler;
}

export type SectionCtx = {
  selected: boolean;
  segments: {
    [key: string]: SegmentCtx;
  };
};
