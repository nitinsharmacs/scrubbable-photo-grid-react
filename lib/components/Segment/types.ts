import type { SegmentConfig, SegmentType } from 'lib/types';

export interface SegmentProps {
  segment: SegmentType;
  map: SegmentConfig;
}
