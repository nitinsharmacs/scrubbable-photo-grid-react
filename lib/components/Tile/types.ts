import type { ImageType, TileType } from 'lib/types';

export interface TileProps {
  tile: TileType;
  image: ImageType;
}
