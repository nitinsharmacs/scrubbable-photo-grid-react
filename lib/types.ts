export type SectionType = {
  sectionId: string;
  totalImages: number;
  segments: SegmentType[];
  header: string;
};

export type SegmentType = {
  segmentId: string;
  header: string;
  images: ImageType[];
};

export type TileType = {
  width: number;
  height: number;
  top: number;
  left: number;
  aspectRatio?: number;
};

export type ImageType = {
  metadata: {
    width: number;
    height: number;
    orientation?: number;
  };
  imageurl: string;
  imageId: string;
};

export type SegmentConfigType = {
  top: number;
  width: number;
  height: number;
  tiles: TileType[];
  headerHeight: number;
};

export type SegmentsMapType = {
  [key: string]: SegmentConfigType;
};

export type SectionConfigType = {
  height: number;
  width?: number;
  top: number;
  visible: boolean;
  index: number;
  segmentsMap: SegmentsMapType;
  headerHeight: number;
};

export type SectionsMapType = {
  [key: string]: SectionConfigType;
};

export type GridConfigType = {
  containerWidth: number;
  sectionMargin: number;
  segmentMargin: number;
  targetRowHeight: number;
};
