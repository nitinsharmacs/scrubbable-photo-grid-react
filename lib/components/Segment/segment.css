.segment {
  background-color: rgb(218, 212, 212);
  position: absolute;
  contain: layout;
}

.segment-header {
}

.segment-header > h3 {
  margin: 0;
}

.segment-content {
  position: relative;
}
