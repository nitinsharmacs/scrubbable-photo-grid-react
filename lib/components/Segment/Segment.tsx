import type { SegmentProps } from 'lib/components/Segment/types';
import './segment.css';
import type { TileType } from 'lib/types';
import Tile from 'lib/components/Tile/Tile';

const Segment = ({ segment, map }: SegmentProps) => {
  return (
    <div
      className='segment'
      id={segment.segmentId}
      style={{
        width: `${map.width}px`,
        height: `${map.height}px`,
        top: `${map.top}px`,
      }}
    >
      <div className='segment-header' style={{ height: map.headerHeight }}>
        <h3>{segment.header}</h3>
      </div>
      <div className='segment-content'>
        {map.tiles.map((tile: TileType, index: number) => (
          <Tile
            key={segment.images[index].imageId}
            tile={tile}
            image={segment.images[index]}
          />
        ))}
      </div>
    </div>
  );
};

export default Segment;
