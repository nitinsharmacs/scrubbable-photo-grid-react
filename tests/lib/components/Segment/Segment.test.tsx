import React from 'react';

import { render } from '@testing-library/react';
import Segment from 'lib/components/Segment/Segment';
import { SegmentConfigType, SegmentType } from 'lib/types';
import { describe, expect, it } from 'vitest';

describe('Segment', () => {
  it('should match snapshot', () => {
    const segment: SegmentType = {
      segmentId: 'sec1',
      header: 'header',
      images: [],
    };

    const map: SegmentConfigType = {
      top: 0,
      width: 100,
      height: 100,
      tiles: [],
    };
    const { container } = render(<Segment segment={segment} map={map} />);
    expect(container).toMatchSnapshot();
  });

  it('should have tiles', () => {
    const segment: SegmentType = {
      segmentId: 'sec1',
      header: 'header',
      images: [
        {
          imageId: 'something',
          imageurl: 'something',
          metadata: {
            width: 100,
            height: 100,
          },
        },
        {
          imageId: 'something2',
          imageurl: 'something',
          metadata: {
            width: 50,
            height: 100,
          },
        },
      ],
    };

    const map: SegmentConfigType = {
      top: 0,
      width: 100,
      height: 100,
      tiles: [
        {
          width: 100,
          height: 100,
          top: 20,
          left: 20,
        },
        {
          width: 100,
          height: 100,
          top: 120,
          left: 120,
        },
      ],
    };

    const { container } = render(<Segment segment={segment} map={map} />);
    expect(container.querySelectorAll('.tile').length).toBe(2);
  });
});
