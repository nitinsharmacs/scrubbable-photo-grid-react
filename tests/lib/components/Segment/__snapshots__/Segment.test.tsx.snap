// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Segment > should match snapshot 1`] = `
<div>
  <div
    class="segment"
    id="sec1"
    style="width: 100px; height: 100px; top: 0px;"
  >
    <div
      class="segment-header"
      style="height: 40px;"
    >
      <h3>
        header
      </h3>
    </div>
    <div
      class="segment-content"
    />
  </div>
</div>
`;
